File: benchmark/comparison/results/blind_ts/14.ts
function filter(lst: number[], pred: (x: number) => boolean): number[] {
  const result: number[] = [];
  for (const x of lst) {
    if (pred(x)) result.push(x);
  }
  return result;
}

function gt3(x: number): boolean { return x > 3; }
function is_even(x: number): boolean { return x % 2 === 0; }

function formatList(lst: number[]): string {
  return "[" + lst.join(", ") + "]";
}

console.log(formatList(filter([1, 2, 3, 4, 5], gt3)));
console.log(formatList(filter([1, 2, 3], is_even)));
console.log(formatList(filter([], gt3)));

File: benchmark/comparison/results/blind_py/14.py
def filter_list(lst, pred):
    return [x for x in lst if pred(x)]

def gt3(x):
    return x > 3

def is_even(x):
    return x % 2 == 0

def fmt(lst):
    return "[" + ", ".join(str(x) for x in lst) + "]"

print(fmt(filter_list([1, 2, 3, 4, 5], gt3)))
print(fmt(filter_list([1, 2, 3], is_even)))
print(fmt(filter_list([], gt3)))
